internals/maintenance_scripts.py

```diff
@@ -344,22 +344,24 @@ def find_trial_stage(self, feature_id: int) -> Stage|None:
         return None
 
       trial_stage_type = STAGE_TYPES_ORIGIN_TRIAL[fe.feature_type]
-      trial_stages = Stage.query(
+      trial_stages: list[Stage] = Stage.query(
           Stage.stage_type == trial_stage_type,
           Stage.feature_id == feature_id).fetch()
       # If there are no OT stages for the feature, we can't associate the
       # trial with any stages.
       if len(trial_stages) == 0:
         logging.info(f'No OT stages found for feature ID: {feature_id}')
         return None
-      # If there is currently more than one origin trial stage for the
-      # feature, we don't know which one represents the given trial.
-      if len(trial_stages) > 1:
+      # Look for a stage that does not already have an origin trial associated
+      # with it.
+      unassociated_trial_stages =  [s for s in trial_stages
+                                    if not s.origin_trial_id]
+      if len(unassociated_trial_stages) > 1:
         logging.info('Multiple origin trial stages found for feature '
                      f'{feature_id}. Cannot discern which stage to associate '
                      'trial with.')
         return None
-      return trial_stages[0]
+      return unassociated_trial_stages[0]
```

Comment:
What if all stages already have a OT associated with it? `unassociated_trial_stages[0]` will probably cause index out of bound

Reply:
One more comment - add a test case for this